tests/crypto/crl_tests.c

```diff
@@ -127,6 +127,19 @@ static bool _year_is_a_leap_year(uint32_t year)
     }
 }
 
+static int32_t _diff_oe_datetime(oe_datetime_t* t1, oe_datetime_t* t2)
+{
+    int32_t dyear = (int32_t)t1->year - (int32_t)t2->year;
+    int32_t dmonth = (int32_t)t1->month - (int32_t)t2->month;
+    int32_t dday = (int32_t)t1->day - (int32_t)t2->day;
+    int32_t dhour = (int32_t)t1->hours - (int32_t)t2->hours;
+    int32_t dmin = (int32_t)t1->minutes - (int32_t)t2->minutes;
+    int32_t dsec = (int32_t)t1->seconds - (int32_t)t2->seconds;
+
+    return (dyear * 31556952) + (dmonth * 2592000) + (dday * 86400) +
```

Comment:
These are date-time values and not cumulative time values, so individual elements in t1 may be less that individual elements in t2, resulting in the addition of negative numbers. If you want to do this, you'd need to do the conversion to cumulative time first, and then the diff

It doesn't necessarily matter in the use case here, but really it seems like you should implement a specific boundary limit function instead of implementing a generic datetime diff function. i.e. in `_test_get_dates()` call `oe_datetime_compare()` and if they are not equal, then you only have to deal with checking that your min + sec difference is within your expected time bound.

Reply:
Dave's suggestion to convert to time_t is probably easier. I'll look into that.